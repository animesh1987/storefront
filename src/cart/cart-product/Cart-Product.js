import React from 'react';
import './Cart-Product.css';

export const CartProduct = (props) => {
  const { product } = props;
  return (
    <tr className="Cart-Product flex flex-row">
      <td className="flex flex-row">
        <img src={`${process.env.PUBLIC_URL}/media/${product.image}`}
          alt={product.title} />
        <div className="flex flex-column">
          <span>{product.brand}</span>
          <span>{product.title}</span>
        </div>
      </td>
      <td className="Cart-Product__actions flex flex-row">
        <span>{product.quantity}</span>
        <div className="flex flex-column">
          <i onClick={() => 
            props.updateProductQuantity({type: 'increment', product})}
            className="material-icons">add</i>
          <i onClick={() => 
            props.updateProductQuantity({type: 'decrement', product})}
            className="material-icons">remove</i>
        </div>
      </td>
      <td>${product.quantity * product.price}.00</td>
      <td>
        <i onClick={() => props.removeProductFromCart(product.id)}
          className="material-icons">close</i>
      </td>
    </tr>
  )
};
