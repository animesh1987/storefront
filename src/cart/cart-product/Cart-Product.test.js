import React from 'react';
import { create } from 'react-test-renderer';
import { CartProduct } from './Cart-Product';
 
it('renders product page correctly', () => {
  const product = {
    "id": "1",
    "title": "Blue Stripe Stoneware Plate",
    "brand": "Kiriko",
    "price": 40,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "blue-stripe-stoneware-plate.jpg"
  };
  const component = create(<CartProduct
      key={product.id}
      product={product} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
