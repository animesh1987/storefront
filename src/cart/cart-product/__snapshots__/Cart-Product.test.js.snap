// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product page correctly 1`] = `
<tr
  className="Cart-Product flex flex-row"
>
  <td
    className="flex flex-row"
  >
    <img
      alt="Blue Stripe Stoneware Plate"
      src="/media/blue-stripe-stoneware-plate.jpg"
    />
    <div
      className="flex flex-column"
    >
      <span>
        Kiriko
      </span>
      <span>
        Blue Stripe Stoneware Plate
      </span>
    </div>
  </td>
  <td
    className="Cart-Product__actions flex flex-row"
  >
    <span />
    <div
      className="flex flex-column"
    >
      <i
        className="material-icons"
        onClick={[Function]}
      >
        add
      </i>
      <i
        className="material-icons"
        onClick={[Function]}
      >
        remove
      </i>
    </div>
  </td>
  <td>
    $
    NaN
    .00
  </td>
  <td>
    <i
      className="material-icons"
      onClick={[Function]}
    >
      close
    </i>
  </td>
</tr>
`;
